homeassistant/components/fortigate/device_tracker.py

```diff
@@ -0,0 +1,110 @@
+"""Device tracker for Fortigate Firewalls."""
+from collections import namedtuple
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_USERNAME, \
+    CONF_PASSWORD, CONF_DEVICES
+from homeassistant.components.device_tracker import DeviceScanner
+from homeassistant.helpers import config_validation as cv
+from homeassistant.components.device_tracker import PLATFORM_SCHEMA
+
+
+from . import DATA_FGT
+
+_LOGGER = logging.getLogger(__name__)
+
+DETECTED_DEVICES = "/monitor/user/detected-device"
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_DEVICES, default=[]):
+        vol.All(cv.ensure_list, [cv.string]),
+})
+
+
+async def async_get_scanner(hass, config):
+    """Validate the configuration and return a Freebox scanner."""
+    scanner = FortigateDeviceScanner(hass.data[DATA_FGT])
+    await scanner.async_connect()
+    return scanner if scanner.success_init else None
+
+
+Device = namedtuple('Device', ['hostname', 'mac'])
+
+
+def _build_device(device_dict):
+    return Device(
+        device_dict['hostname'],
+        device_dict['mac'])
+
+
+class FortigateDeviceScanner(DeviceScanner):
+    """Queries the Fortigate firewall."""
+
+    def __init__(self, hass_data):
+        """Initialize the scanner."""
+        self.last_results = {}
+        self.success_init = False
+        self.connection = hass_data['fgt']
+        self.devices = hass_data['devices']
+
+    def get_results(self):
+        """Get the results from the fortigate."""
+        results = self.connection.get(
+            DETECTED_DEVICES, "vdom=root")[1]['results']
+
+        ret = []
+        for result in results:
+            try:
+                # some device does not have a hostname
+                assert 'hostname' in result
```

Comment:
Don't use `assert` to replace `if` check.
```py
if 'hostname' not in result:
    continue
...
```

Reply:
fixed, thanks